From e4d6534d0c544ed9857947257645663edc2fe3a3 Mon Sep 17 00:00:00 2001
From: Eric Anholt <eric@anholt.net>
Date: Mon, 4 Apr 2016 14:25:59 -0700
Subject: [PATCH 079/101] drm/vc4: Fix NULL deref in HDMI init error path
Git-commit: 4ef3e2e380cff9f6dfbffc0e63188bd681548cd3
Patch-mainline: Queued in subsystem maintainer repository
Git-repo: https://github.com/anholt/linux.git
References: FATE#319481

If you make it here other than through err_destroy_encoder, vc4->hdmi
is still NULL.

Signed-off-by: Eric Anholt <eric@anholt.net>
Signed-off-by: Alexander Graf <agraf@suse.de>

---
 drivers/gpu/drm/vc4/vc4_hdmi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/vc4/vc4_hdmi.c b/drivers/gpu/drm/vc4/vc4_hdmi.c
index 85224e5..524eae9 100644
--- a/drivers/gpu/drm/vc4/vc4_hdmi.c
+++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
@@ -573,7 +573,7 @@ err_unprepare_hsm:
 err_unprepare_pix:
 	clk_disable_unprepare(hdmi->pixel_clock);
 err_put_i2c:
-	put_device(&vc4->hdmi->ddc->dev);
+	put_device(&hdmi->ddc->dev);
 
 	return ret;
 }
-- 
1.8.5.6

